compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/kotlin/KSClassDeclarationImpl.kt

```diff
@@ -60,8 +60,13 @@ class KSClassDeclarationImpl private constructor(val ktClassOrObject: KtClassOrO
         val propertiesFromConstructor = primaryConstructor?.parameters
             ?.filter { it.isVar || it.isVal }
             ?.map { KSPropertyDeclarationParameterImpl.getCached((it as KSValueParameterImpl).ktParameter) } ?: emptyList()
-        val result = ktClassOrObject.declarations.getKSDeclarations().toMutableList()
+        val result = ktClassOrObject.declarations.getKSDeclarations().toMutableList<KSDeclaration>()
         result.addAll(propertiesFromConstructor)
+        primaryConstructor?.let { primaryConstructor: KSDeclaration ->
```

Comment:
How about using the origin of `primaryConstructor` to determine if it needs to be added into the return result?

Reply:
you mean `if (primaryConstructor.origin == Origin.KOTLIN) {` ? if so, it seems to be working fine (passes tests and behaves the same for tests we have).